limiter.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package rate
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// Limiter is used to determine if a request for a given resource and action
+// should be allowed.
+// TODO: expand this doc
+type Limiter struct {
+	limits map[string]*Limit
+
+	sync.RWMutex
```

Comment:
When would a client of Limiter be expected to call Lock and Unlock?

Reply:
We don't expect that, I'll make this un-exported.